src/main/java/org/folio/circulation/support/utils/DateFormatUtil.java

```diff
@@ -0,0 +1,557 @@
+package org.folio.circulation.support.utils;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.temporal.ChronoField.HOUR_OF_DAY;
+import static java.time.temporal.ChronoField.MILLI_OF_SECOND;
+import static java.time.temporal.ChronoField.MINUTE_OF_HOUR;
+import static java.time.temporal.ChronoField.NANO_OF_SECOND;
+import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
+import static org.folio.circulation.support.utils.DateTimeUtil.normalizeDate;
+import static org.folio.circulation.support.utils.DateTimeUtil.normalizeDateTime;
+import static org.folio.circulation.support.utils.DateTimeUtil.normalizeZone;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.ISODateTimeFormat;
+
+/**
+ * A utility for common date time formatting.
+ */
```

Comment:
Can you copy the key ideas explained in the Purpose/Approach/Learning sections of this pull request's description into the DateFormatUtil class level javadoc?

Reply:
Done.